vpr/src/base/clustered_netlist.cpp

```diff
@@ -306,3 +306,56 @@ bool ClusteredNetlist::validate_net_sizes_impl(size_t num_nets) const {
     }
     return true;
 }
+
+/**
+ * @brief Finds a block where the block's name matches to the provided
+ *        input string pattern. 
+ *        The intented use is to find the block id of a 
+ *        hard block without knowing its name in the netlist. Instead
+ *        a pattern can be created that we know the block's name will
+ *        match to. Generally, we expect the pattern to be constructed
+ *        using the block's module name in the HDL design, since we can
+ *        expect the netlist name of the block to include the module
+ *        name within it.
+ * 
+ *        For example, suppose a RAM block was named in the netlist as
+ *        "top|alu|test_ram|out". The user instantiated the ram module
+ *        in the HDL design as "test_ram". So instead of going through 
+ *        the netlist and finding the ram block's full name, this
+ *        function can be used by just providing the string pattern as
+ *        ".*test_ram.*". We know that the module name should be somewhere
+ *        within the string, so the pattern we provide says that the netlist
+ *        name of the block contains arbritary characters then the module name
+ *        and then some other arbritary characters after. This pattern will
+ *        then be used to match to the block in the netlist. 
+ * 
+ *        This function runs in linear time (O(N)) as it goes over all the 
```

Comment:
Update this to say you go through the cluster_block_candidates, and are O(N) in that.  The cluster_block_candidates could be the whole netlist, or a subset of the correct type, or anything else.

Reply:
(Unless this is redudant with the .h, in which case just shorten/ensure it's accurate).

Reply:
Done